icepyx/core/query.py

```diff
@@ -111,22 +111,29 @@ class Query:
 
     def __init__(
         self,
-        dataset=None,
+        product=None,
         spatial_extent=None,
         date_range=None,
         start_time=None,
         end_time=None,
         version=None,
         cycles=None,
         tracks=None,
-        files=None,
+        files=None,  # NOTE: if you end up implemeting this feature here, use a better variable name than "files"
+        dataset=None,
     ):
 
         # warnings.filterwarnings("always")
         # warnings.warn("Please note: as of 2020-05-05, a major reorganization of the core icepyx.query code may result in errors produced by now depricated functions. Please see our documentation pages or example notebooks for updates.")
 
+        if dataset != None:
+            warnings.filterwarnings("always")
+            warnings.warn(
+                "In line with most common usage, the 'dataset' input argument has been replaced by 'product'."
+            )
+
```

Comment:
Thanks for this PR @JessicaS11, it looks great! Just some small suggestions. Since we have changed all `dataset` into `product`, do we still need to keep `dataset` parameter in `Query` class here? I think we can just update the example notebooks and add a note that we have changed `dataset` into `product`, I reckon most users won't use keyword argument `product = 'ATL03'` to specify the product they want, but would directly pass product name as the first positional argument as the case used in example notebook :`region_a = ipx.Query(short_name, spatial_extent, date_range)`.

Reply:
Thanks for the review, @icetianli, and for catching these problems. I fixed the unintentional non-icepyx library removals of dataset you noted. I suspect you're right that most users aren't using the keyword, so I removed the warning from the `Query.__init__` and added a note, as you suggested. Your comment also made me realize that I didn't update the example notebooks, so I went through and made those changes as well. Since there had been a `dataset` property on the query object that was replaced by a `product` property, I went ahead and made a `dataset` property that has the deprecation warning instead.

Reply:
Thanks for the changes! Regarding the failed pytest on test_visualization.py: `cannot import name 'ABCIndexClass' from 'pandas.core.dtypes.generic'`, I found that it is because the [`holoviews`](https://github.com/holoviz/holoviews/blob/master/holoviews/core/util.py) isn't compatible with the latest `pandas` - version 1.3.0 released four days ago - they have now changed ['ABCIndexClass' into 'ABCIndex' ](https://github.com/pandas-dev/pandas/commit/eb67d87b6b133242e41293054238c17d54a69bd9#diff-d19c748aa6d9ba07f1a0a428729a220e227f8616cddce4a5e202ad793a04d66b) so holoviews cannot import them anymore. To solve this issue, could we specify the default pandas version to be 1.2.5?